[DROOLS-7164] Drools 8 docs : Rule Unit docs review and update (#4704)
kiegroup/drools: drools-docs/src/modules/ROOT/pages/KIE/BuildDeployUtilizeAndRun/_KIEBuilding-section.adoc

[[_enginedependency]]
== Engine dependency

> {PRODUCT} has 3 types of engine dependency that aggregates a required collection of dependencies.

Reviewer: ```suggestion
{PRODUCT} has 3 types of engine dependency that aggregate a required collection of dependencies.
```

> *`drools-ruleunits-engine`* is the standard engine for rule units that enables executable model.

Reviewer: ```suggestion
*`drools-ruleunits-engine`* is the standard engine for rule units that enables an executable model.
```

> *`drools-engine`* is the standard engine for traditional DRL syntax that enables executable model.

Reviewer: ```suggestion
*`drools-engine`* is the standard engine for traditional DRL syntax that enables an executable model.
```

> *`drools-engine-classic`* is the old engine that utilizes an MVEL interpreter.  The `drools-engine-classic` and `drools-mvel` are now *deprecated*, so use `drools-ruleunits-engine` or `drools-engine` as possible.

Reviewer: ```suggestion
*`drools-engine-classic`* is the old engine that utilizes an MVEL interpreter.  The `drools-engine-classic` and `drools-mvel` are now *deprecated*, so use `drools-ruleunits-engine` or `drools-engine` instead.
```
